### ISSUE TITLE
"No newline at end of file" icon is not exposed to screenreaders

### ISSUE THREAD

**Issue by outofambit** [desktop/desktop#6494]
<img width="535" alt="no newline at end of file icon screenshot" src="https://user-images.githubusercontent.com/964912/50457138-4860b400-090e-11e9-97e7-76c6aaff2eab.png">

The "No newline at end of file" icon (shown above) does not have alt text for screen readers to read.

The relevant code for this markup lives at:

https://github.com/desktop/desktop/blob/1b69c2f04bd55afbe363821dccd0a3966c40e509/app/src/ui/diff/text-diff.tsx#L428

We'll want to add an alt tag to this markup. The alt text should probably read "No newline at end of file".

**Comment by shiftkey** [comment]
@rmsmani apologies for the delay in replying. I think we need to figure out #6495 first before we can do this, and I'm going to follow up there on some questions I have about the changes that need to be made.

**Comment by shiftkey** [comment]
I had a look into some of the history about this change and we render a `title` tag into the parent container, since way back in  https://github.com/desktop/desktop/commit/396fea92dc0e28cc3eca54655c64b39cdb4a03ec

https://github.com/desktop/desktop/blob/96eb4a8894f048cb997d86ae413f3d0bfc33be38/app/src/ui/diff/text-diff.tsx#L422-L431

I'm curious why we didn't use `title` in `Octicon` (which was available) in https://github.com/desktop/desktop/pull/1320 - @niik can you recall?

**Comment by shiftkey** [comment]
Dropping the labels on this while it's blocked on #6495

**Comment by billygriffin** [comment]
@rmsmani Hi! Since #6495 shipped, would you like to take another stab at this issue? If not, no problem, just let us know!

**Comment by rmsmani** [comment]
Sure, I can take some other...

With Regards
M.Mani
+61 432 461 087

**Comment by jmhammock** [comment]
I would like to pick this issue up. It seems that a lot has changed since this issue was created. I have a fix working locally, but it might be good to confirm some details before I submit a PR.
It looks like the Octicon component is no longer being used to render the noNewline svg and instead an HTMLSpanElement is being built and passed to Doc.setBookmark which requires the widget to be of type HTMLElement. 
https://github.com/desktop/desktop/blob/2870bfb005e569d37c142d5c17f4f7138a44a20c/app/src/ui/diff/text-diff.tsx#L211-L218
If everyone is OK with not using the Ociton component, then I will submit a PR. If the Octicon component is to be used, then there is more work to do.

**Comment by tierninho** [comment]
@jmhammock Thanks for the potential fix. Since you already have a PR ready, I'd suggest submitting it with your questions included. It will be easier to get feedback this way.
